README.md

```diff
@@ -9,91 +9,168 @@ A libGDX cross platform video rendering extension
 
-#### Repositories
+### Repositories
 
 * **Official**  For official releases, use https://oss.sonatype.org/content/repositories/releases
 * **Snapshots** For snapshot builds, use https://oss.sonatype.org/content/repositories/snapshots
 
-#### Gradle dependency declarations
-Don't forget to replace _0.0.1_ with the current version!
+### Gradle dependency declarations
 ##### Core:
 ```groovy
-implementation "com.badlogicgames.gdx-video:gdx-video:0.0.1"
+implementation "com.badlogicgames.gdx-video:gdx-video:$gdxVideoVersion"
 ```
+
 ##### Desktop:
 ```groovy
-implementation "com.badlogicgames.gdx-video:gdx-video-lwjgl:0.0.1"
+implementation "com.badlogicgames.gdx-video:gdx-video-lwjgl3:$gdxVideoVersion"
+// or (when using legacy LWJGL2):
+implementation "com.badlogicgames.gdx-video:gdx-video-lwjgl:$gdxVideoVersion"
 ```
-or
+
+##### Android:
 ```groovy
-implementation "com.badlogicgames.gdx-video:gdx-video-lwjgl3:0.0.1"
+implementation "com.badlogicgames.gdx-video:gdx-video-android:$gdxVideoVersion"
 ```
 
-##### Android:
+##### iOS (with RoboVM):
 ```groovy
-implementation "com.badlogicgames.gdx-video:gdx-video-android:0.0.1"
+implementation "com.badlogicgames.gdx-video:gdx-video-robovm:$gdxVideoVersion"
 ```
 
 ##### Html:
 
 ```groovy
-implementation "com.badlogicgames.gdx-video:gdx-video:0.0.1:sources"
-implementation "com.badlogicgames.gdx-video:gdx-video-gwt:0.0.1"
-implementation "com.badlogicgames.gdx-video:gdx-video-gwt:0.0.1:sources"
+implementation "com.badlogicgames.gdx-video:gdx-video:$gdxVideoVersion:sources"
+implementation "com.badlogicgames.gdx-video:gdx-video-gwt:$gdxVideoVersion"
+implementation "com.badlogicgames.gdx-video:gdx-video-gwt:$gdxVideoVersion:sources"
 ```
 You also need to add the following file to your GdxDefinition.gwt.xml in your html project:
 `````xml
 <inherits name="com.badlogic.gdx.video.gdx_video_gwt" />
 `````
-#### Maven
-Don't forget to replace _0.0.1_ with the current version!
-
-Core:
-```xml
-<dependency>
-  <groupId>com.badlogicgames.gdx-video</groupId>
-  <artifactId>gdx-video</artifactId>
-  <version>0.0.1</version>
-</dependency>
+
+## Playing video
+
+First, get a video player for the current platform using
+
+```java
+VideoPlayer player = VideoPlayerCreator.createVideoPlayer();
 ```
-Desktop:
-```xml
-<dependency>
-  <groupId>com.badlogicgames.gdx-video</groupId>
-  <artifactId>gdx-video-lwjgl</artifactId>
-  <version>0.0.1</version>
-</dependency>
+
+Then, open a video from your game's assets.
+
+```java
+FileHandle file = Gdx.files.internal("video.webm");
+player.play(file);
 ```
-or
-```xml
-<dependency>
-  <groupId>com.badlogicgames.gdx-video</groupId>
-  <artifactId>gdx-video-lwjgl3</artifactId>
-  <version>0.0.1</version>
-</dependency>
+
+The file gets loaded and starts playing as soon as the first frames
+are decoded. If you want to load a video ahead of time, `pause()` 
+immediately and `resume()` playback later.
```

Comment:
I think that's not a good design to require a hack like this. I think a better design would be to introduce two methods `preload(FileHandle file)` and `play()`. In `play (FileHandle file)` we can just call preload and play (or we remove the method). What do you think?

Reply:
Thanks! I also think separate functions to load and play make more sense.

For the individual `play` function, how should it behave?
1. *play now if the video is loaded, or return false*
2. *play now. If the video is not ready yet, start playing as soon as the video is loaded*

In order to keep the existing method, we need to implement option 2.

Do you think I should make this change as part of this PR too or should this be made separately later?

Reply:
I think in general, it would be nice to query the state, with direct queries or callbacks so we know when its ready to play, and then its up to the user to decide when they want to call play. 
`preload` with callback so we can get an event when its ready, play (that can probably warn/error) if attempted to play before ready state achieved.

Reply:
> Decoding still happens after pause?

On Desktop and iOS it does, until buffers are full. On Android, it seems implementation dependent. Don't know about Web / GWT.

> I think in general, it would be nice to be able to query the state, with direct queries or callbacks so we know when its ready to play, and then its up to the user to decide when they want to call play. `preload` with callback so we can get an event when its ready, play (that can probably warn/error) if attempted to play before ready state achieved.
> 
> Might not be any harm in being able call play in any state, as long as its documented its not guaranteed to actually start playing until its ready.

I think a query `isLoaded()`, `isReady()` or similar is useful. But even with a query, I like the idea of `preload` with a callback once the video is ready to play.

Reply:
You can see my half-baked preload solution for GWT at #74, for what little that's worth.

Reply:
I updated that part of the README to tell users that preloading is not officially supported yet.

I still mention pausing immediately as possible workaround, and made sure we can deal with that on all platforms. Thanks @Frosty-J for your input on GWT.